### PR TITLE
Use multi-stage build for Docker

### PR DESCRIPTION
**Pull request by mortenlj**
The setup here is really quite simple:

* First we create a stage named `common`, where we install anything that is needed both while building and at runtime. In our case, it's just libyaml.
* Second, we create a stage named `build`, based on `common`. Here we install build-tools, needed headers etc. Then we build binary wheels of all our dependencies and our application.
* Last, we create a stage named `production`, based on `common` again. This gets rid of all the tools and stuff needed to build the binaries. We then copy the binary wheels from `build` and use those to install the application.

### CODE REVIEW

**Comment by mortenlj** [comment]
The codacy issues comes from Hadolint being outdated, as all three issues are false negatives. I'm going to merge anyway.

File: setup.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def version():
     author_email="fiaas@googlegroups.com",
     version=version(),
     packages=find_packages(exclude=("tests",)),
-    zip_safe=False,
+    zip_safe=True,
```
Comment (birgirst) [inline setup.py]: This breaks the frontend application as the templates cannot be loaded.

Comment (mortenlj) [reply setup.py]: That's weird. It shouldn't... :thinking: 
:man_shrugging: Easy enough to flip the flag back though, I'll fix.

Comment (mortenlj) [reply setup.py]: #5

**Pull request merged**